### ISSUE TITLE
Migrate to IDF v5.3

### ISSUE THREAD

**Issue by forGGe** [maxlab-io/tokay-lite-sdk#4]
IDF v5.0 has issues building on MacOS it seems:

```
CMake Error at /Users/redacted/esp/esp-idf/tools/cmake/kconfig.cmake:176 (message):
  Failed to run confgen.py
  (/Users/redacted/.espressif/python_env/idf5.4_py3.11_env/bin/python;/Users/redacted/esp/esp-idf/tools/kconfig_new/confgen.py;--list-separator=semicolon;--kconfig;/Users/redacted/esp/esp-idf/Kconfig;--sdkconfig-rename;/Users/redacted/esp/esp-idf/sdkconfig.rename;--config;/Users/redacted/tokay-lite-sdk/webserver/sdkconfig;--env-file;/Users/redacted/tokay-lite-sdk/webserver/build/config.env).
  Error 1
Call Stack (most recent call first):
  /Users/redacted/esp/esp-idf/tools/cmake/build.cmake:590 (__kconfig_generate_config)
  /Users/redacted/esp/esp-idf/tools/cmake/project.cmake:440 (idf_build_process)
  CMakeLists.txt:11 (project)


-- Configuring incomplete, errors occurred!
FAILED: build.ninja 
/opt/homebrew/Cellar/cmake/3.30.2/bin/cmake --regenerate-during-build -S/Users/redacted/tokay-lite-sdk/webserver -B/Users/redacted/tokay-lite-sdk/webserver/build
ninja: error: rebuilding 'build.ninja': subcommand failed
ninja failed with exit code 1, output of the command is in the /Users/redacted/tokay-lite-sdk/webserver/build/log/idf_py_stderr_output_78960 and /Users/redacted/tokay-lite-sdk/webserver/build/log/idf_py_stdout_output_78960
redacted@MacBook-Pro---KM tokay-lite-sdk % pip3 install future      
Requirement already satisfied: future in /Users/redacted/.espressif/python_env/idf5.0_py3.11_env/lib/python3.11/site-packages (0.18.2)
redacted@MacBook-Pro---KM tokay-lite-sdk % pip3 install -U future
Requirement already satisfied: future in /Users/redacted/.espressif/python_env/idf5.0_py3.11_env/lib/python3.11/site-packages (0.18.2)
Collecting future
  Using cached future-1.0.0-py3-none-any.whl.metadata (4.0 kB)
Using cached future-1.0.0-py3-none-any.whl (491 kB)
Installing collected packages: future
  Attempting uninstall: future
    Found existing installation: future 0.18.2
    Uninstalling future-0.18.2:
      Successfully uninstalled future-0.18.2
Successfully installed future-1.0.0
redacted@MacBook-Pro---KM tokay-lite-sdk % pip3 install future      
Requirement already satisfied: future in /Users/redacted/.espressif/python_env/idf5.0_py3.11_env/lib/python3.11/site-packages (1.0.0)
redacted@MacBook-Pro---KM tokay-lite-sdk % idf.py clean -C webserver
/Users/redacted/esp/esp-idf/tools/check_python_dependencies.py:12: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  import pkg_resources
The following Python requirements are not satisfied:
future<0.18.3,>=0.15.2 - (future 1.0.0 (/Users/redacted/.espressif/python_env/idf5.0_py3.11_env/lib/python3.11/site-packages), Requirement.parse('future<0.18.3,>=0.15.2'))
To install the missing packages, please run "install.sh"
Diagnostic information:
    IDF_PYTHON_ENV_PATH: /Users/redacted/.espressif/python_env/idf5.0_py3.11_env
    Python interpreter used: /Users/redacted/.espressif/python_env/idf5.0_py3.11_env/bin/python
Constraint file: /Users/redacted/.espressif/espidf.constraints.v5.0.txt
Requirement files:
 - /Users/redacted/esp/esp-idf/tools/requirements/requirements.core.txt
Python being checked: /Users/redacted/.espressif/python_env/idf5.0_py3.11_env/bin/python

ESP-IDF v5.0

```

The build v5.3 passes that stage and fails when building tokay-lite code itself.

**Comment by DeimosHall** [comment]
Hi, I built the project using IDF v5.0 with no problems on a Macbook Air M1, everything works fine. However, the project needs some fixes in sdkconfig when using IDF v5.3 to do the build, but with this version I had problems so I moved to v5.0

**Comment by forGGe** [comment]
> Hi, I built the project using IDF v5.0 with no problems on a Macbook Air M1, everything works fine. However, the project needs some fixes in sdkconfig when using IDF v5.3 to do the build, but with this version I had problems so I moved to v5.0

weird. it feels now like some leftovers in the system are causing issues with the build. 

anyway, https://github.com/maxlab-io/tokay-lite-sdk/pull/7 is merged, v5.3 should be usable starting from now
